Source/iOS/AppLovin-MAX-Godot-Plugin/AppLovinMAXGodotPlugin.mm

```diff
@@ -355,6 +382,113 @@ void applovin_max_godot_plugin_deinit()
     return !ALUtils.simulator;
 }
 
+#pragma mark - Consent Flow
+
+void AppLovinMAXGodotPlugin::set_terms_and_privacy_policy_flow_enabled(bool enabled)
+{
+    _sdk.settings.termsAndPrivacyPolicyFlowSettings.enabled = enabled;
+}
+
+void AppLovinMAXGodotPlugin::set_privacy_policy_url(String url_string)
+{
+    _sdk.settings.termsAndPrivacyPolicyFlowSettings.privacyPolicyURL = [NSURL URLWithString: NSSTRING(url_string)];
+}
+
+void AppLovinMAXGodotPlugin::set_terms_of_service_url(String url_string)
+{
+    _sdk.settings.termsAndPrivacyPolicyFlowSettings.termsOfServiceURL = [NSURL URLWithString: NSSTRING(url_string)];
+}
+
+void AppLovinMAXGodotPlugin::set_consent_flow_debug_user_geography(String user_geography)
+{
+    _sdk.settings.termsAndPrivacyPolicyFlowSettings.debugUserGeography = (ALConsentFlowUserGeography) NSSTRING(user_geography).intValue;
+}
+
+void AppLovinMAXGodotPlugin::show_cmp_for_existing_user()
+{
+    if ( !_isSdkInitialized )
+    {
+        NSLog(@"[%@] Failed to show CMP for existing user - please ensure the AppLovin MAX Godot Plugin has been initialized by calling 'AppLovinMAX.initialize()'!", TAG);
+        return;
+    }
+    
+    [_sdk.cmpService showCMPForExistingUserWithCompletion:^(ALCMPError * _Nullable error) {
+        
+        Dictionary cmp_error = Dictionary();
+
+        if ( error )
+        {
+            cmp_error["code"] = (int) error.code;
+            cmp_error["message"] = GODOT_STRING(error.message);
+            cmp_error["cmpCode"] =  (int) error.cmpCode;
+            cmp_error["cmpMessage"] = GODOT_STRING(error.cmpMessage);
+        }
+
+        call_deferred("emit_signal", AppLovinMAXSignalCmpOnShowForExistingUser, cmp_error);
+   }];
+}
+
+bool AppLovinMAXGodotPlugin::has_supported_cmp()
+{
+    return [_sdk.cmpService hasSupportedCMP];
+}
+
+#pragma mark - Segment Targeting
+
+void AppLovinMAXGodotPlugin::add_segment(int key, Array segment_values)
+{
+    if ( _isSdkInitialized )
+    {
+        NSLog(@"Segment must be added before calling 'AppLovinMAX.initialize()'!");
+        return;
+    }
+    
+    NSMutableArray<NSNumber *> *ns_number_array = [NSMutableArray arrayWithCapacity: segment_values.size()];
+
+    for ( int i = 0; i < segment_values.size(); i++ )
+    {
+        Variant item = segment_values[i];
+        if ( item.get_type() == Variant::INT )
+        {
+            int intVal = item;
+            [ns_number_array addObject: @(intVal)];
+        }
```

Comment:
I think we should log a warning if the type is not what we expect. Though we don't need another check for the actual numeric value since we already do it in native SDKs here: https://github.com/AppLovin/iOS-SDK/blob/master/sdk/Public/MASegment.m#L46

Reply:
Added a log for a warning since type checking is needed.